src/playground/infinite-list/infinite-news-list.component.ts

```diff
@@ -0,0 +1,124 @@
+import { Component, ViewChildren, ElementRef, QueryList, OnInit, Inject, PLATFORM_ID, OnDestroy } from '@angular/core';
+import { isPlatformBrowser } from '@angular/common';
+import { ActivatedRoute, Router } from '@angular/router';
+import { take, filter, map } from 'rxjs/operators';
+import { NbListItemComponent, NbLayoutScrollService, NB_WINDOW } from '@nebular/theme';
+import { getElementHeight } from '@nebular/theme/components/helpers';
+import { NewsService } from './news.service';
+
+@Component({
+  template: `
+    <nb-card>
+      <div [nbSpinner]="loadingPrevious"></div>
+      <nb-list
+        nbInfiniteList
+        listenWindowScroll
+        [threshold]="500"
+        (topThreshold)="loadPrevious()"
+        (bottomThreshold)="loadNext()"
+        [nbListPager]="pageSize"
+        [startPage]="startPage"
+        (pageChange)="updateUrl($event)">
+        <nb-list-item *ngFor="let newsPost of news">
+          <nb-news-post [post]="newsPost"></nb-news-post>
+        </nb-list-item>
+        <nb-list-item *ngFor="let _ of placeholders">
+          <nb-news-post-placeholder></nb-news-post-placeholder>
+        </nb-list-item>
+      </nb-list>
+    </nb-card>
+  `,
+  styleUrls: [ 'infinite-news-list.component.scss' ],
+})
+export class NbInfiniteNewsListComponent implements OnInit, OnDestroy {
+
+  news = [];
+  placeholders = [];
+  pageSize = 10;
+  startPage: number;
+  pageToLoadNext: number;
+  loadingNext = false;
+  loadingPrevious = false;
+  initialScrollRestoration: ScrollRestoration;
+
+  @ViewChildren(NbListItemComponent, { read: ElementRef }) listItems: QueryList<ElementRef<Element>>;
+
+  constructor(
+    private newsService: NewsService,
+    private router: Router,
+    private route: ActivatedRoute,
+    private scrollService: NbLayoutScrollService,
+    @Inject(PLATFORM_ID) private platformId,
+    @Inject(NB_WINDOW) private window,
+  ) {
+    if (isPlatformBrowser(this.platformId) && this.window.history.scrollRestoration) {
+      this.initialScrollRestoration = window.history.scrollRestoration;
+      history.scrollRestoration = 'manual';
+    }
+  }
+
+  ngOnInit() {
+    const { page } = this.route.snapshot.queryParams;
+    this.startPage = page ? Number.parseInt(page, 10) : 1;
+    this.pageToLoadNext = this.startPage;
+  }
+
+  ngOnDestroy() {
+    if (this.initialScrollRestoration) {
+      this.window.history.scrollRestoration = this.initialScrollRestoration;
+    }
+  }
+
+  updateUrl(page) {
+    const queryParams = { ...this.route.snapshot.queryParams, page };
```

Comment:
we might not need to merge it here as `navigate` can do this internally https://angular.io/api/router/NavigationExtras#queryParamsHandling